src/v2/view-builder/views/signin/SignInWithDeviceOption.js

```diff
@@ -30,13 +30,26 @@ export default View.extend({
       icon: 'okta-verify-authenticator',
       title: loc('oktaVerify.button', 'login'),
       click () {
-        if (deviceChallenge.challengeMethod && deviceChallenge.challengeMethod === Enums.UNIVERSAL_LINK_CHALLENGE) {
+        const isUVapproach = deviceChallenge.challengeMethod &&
+          deviceChallenge.challengeMethod === Enums.UNIVERSAL_LINK_CHALLENGE;
+        if (isUVapproach) {
+          // launch the Okta Verify app
           Util.redirect(deviceChallenge.href);
         }
         if (this.options.isRequired) {
           appState.trigger('saveForm', this.model);
         } else {
-          appState.trigger('invokeAction', Enums.LAUNCH_AUTHENTICATOR_REMEDIATION_NAME);
+          // OKTA-350084
+          // For the universal link (iOS) approach,
+          // we need to 1. launch the Okta Verify app
+          // and 2. take the enduser to the next step right away
+          // In Safari, when Okta Verify app is not installed, step 1 responds with error immediately,
+          // then step 2 will respond with error.
+          // To avoid showing the error right before switching to the next UI,
+          // wait for 500 milliseconds before invoking step 2
+          setTimeout(() => {
```

Comment:
I personally don't like introducing setTimeout that blocks the IO.. Can you give me more details about this ?

Reply:
@pradeepdewda-okta I left the reason in the comment. Basically the problem is in step 2 where we switch the user to the next SIW screen. If the step 1 has problem, the step 2 will return error and show on the UI. Basically we want to give it time before we know if the step 1 is okay. If it is okay (the Okta Verify app is installed), the enduser will be temporarily redirected to login.okta.com. So that the step 2 will not even be invoked. Ideally we won't want to use time out, but since we can't get the status of the step 1, this is what we think the best workaround is. Let me know if you have better ideas.

Reply:
What if 500ms is not enough for slower internet connections ? Do you explore polling instead ?

Reply:
That's possible too. But base on the testing we did, 500 is ideal that it does not affect user experience. Also, worst case -- if the enduser gets to the step 2, there's cancel link they can click on to get out of it. This case is very unlikely to happen, we set up a very unrealistic sign on rule just to reproduce the issue.

Reply:
To meet your deadline, i think it would be fine. But i would recommend to circle back to this issue and open a ticket to explore Polling and reimplement it.

I see this view is used at two places. You tested both right ?

Reply:
To be clear, the error is not from the polling. It is from the step 2 `/idp/idx/authenticators/okta-verify/launch` where switching to the next screen of polling happens. Maybe we can look into if there's improvement we can do in the launch api in order for better UX. cc @sshen-okta

Reply:
@jhe-okta @sshen-okta Yes since we have polling framework, we can leverage that. If there is any need to make certain API tweaks to support polling for this use case, we should do that.

Reply:
This is NOT about the polling API.

Reply:
@jhe-okta May be i miss understood. You says we go to step 2 when we can confirm the step 1 was okay.. So to confirm step 1 was okay or not we are waiting for sometime because server takes time to return the value ? Is that right ?

Reply:
If we proceed to step 2 right after step 1 without delay, when step 1 fails, step 2 will respond with error which we show in the UI and it is bad UX. When step 1 succeeds, the enduser will be taken out of the SIW. The time out is to make sure that the redirect happens first before the step 2, so that the UI does not display error. Again, this case is not pratical, it depends on an unrealistic sign on rule to be set up first.

Reply:
Okay. One more thing i want to clear. If Step 1 fails then expectation is still to show Step 2 screen? But introducing delay would not cause error to show up ?

Also can you add this `500` constant to `../../utils/Constants`

Reply:
@jhe-okta can you explore if we can use some kind of promise to resolve first before executing step 2 ?

Reply:
@pradeepdewda-okta updated to use the `Util.callAfterTimeout`.